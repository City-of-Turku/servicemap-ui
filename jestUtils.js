import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });