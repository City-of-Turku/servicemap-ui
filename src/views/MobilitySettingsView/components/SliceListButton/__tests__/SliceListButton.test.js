// Link.react.test.js
import React from 'react';
import SliceList from '../index';
import { getRenderWithProviders } from '../../../../../../jestUtils';
import finnishTranslations from '../../../../../i18n/fi';

const mockProps = {
  openList: true,
  itemsToShow: 4,
  routes: [
    {
      route1: {
        name: 'reitti 1',
      },
      route2: {
        name: 'reitti 2',
      },
    },
  ],
  setItemsToShow: jest.fn(),
};

const renderWithProviders = getRenderWithProviders({});

describe('<SliceList />', () => {
  it('should match snapshot', () => {
    const { container } = renderWithProviders(<SliceList {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<SliceList {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual(finnishTranslations['mobilityPlatform.menu.list.showMore']);
  });

  it('does contain aria-label attribute', () => {
    const { container } = renderWithProviders(<SliceList {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].getAttribute('aria-label')).toEqual(finnishTranslations['mobilityPlatform.menu.list.showMore']);
  });
});
