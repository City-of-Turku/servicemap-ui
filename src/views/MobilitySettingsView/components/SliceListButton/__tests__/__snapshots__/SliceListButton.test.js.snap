// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> should match snapshot 1`] = `
<div>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text injectIntl(SliceListButton)-button-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <p
          aria-label="Näytä lisää reittejä"
          class="MuiTypography-root MuiTypography-body2"
        >
          Näytä lisää reittejä
        </p>
      </span>
    </button>
  </div>
</div>
`;
